lib/js/modules/z.thumbviewer.js

```diff
@@ -0,0 +1,159 @@
+/* thumbviewer.js
+----------------------------------------------------------
+
+Adapted z.imageviewer.js to be able to work with tablets and
+phones.
+ 
+http://www.apache.org/licenses/LICENSE-2.0
+ 
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+---------------------------------------------------------- */
+
+$.widget("ui.thumbviewer", 
+{
+    _init: function() { 
+        let image = this.element;
+
+        /* Defer the init to when the image is loaded */
+        if(image[0].complete) {
+            this.init()
+        } else {
+            $(image[0]).on("load", this.init.bind(this));
+        }
+    },
+	
+    init: function() {
+        var image = this.element;
+        var loadImageFun = this.loadImage.bind(this);
+
+        this.bigImg = $('<img/>').hide();
```

Comment:
Worth wrapping in a `<picture>`, for more styling options in the future?

Reply:
Sure. We need something which works for the whole site. This hooks into the media publishing part of zotonic. See: http://docs.zotonic.com/en/latest/developer-guide/media.html. It uses `{% media ... %}` and `{% image ...%}` tags to help with placing media on pages. Normally this places the img, picture, figure tags. New functionality is also coming here (srcset). It is probably a good idea to have a strategy which works across the whole site, and then hook js-viewers like this into it. This is a very simplistic one.